front/components/editor/input_bar/MentionDropdown.tsx

```diff
@@ -19,165 +20,308 @@ import type {
   MentionDropdownOnKeyDown,
   MentionDropdownProps,
 } from "@app/components/editor/input_bar/types";
+import { useConversationParticipants } from "@app/lib/swr/conversations";
 import { useMentionSuggestions } from "@app/lib/swr/mentions";
+import { useUser } from "@app/lib/swr/user";
 import { classNames } from "@app/lib/utils";
 
 export const MentionDropdown = forwardRef<
   MentionDropdownOnKeyDown,
   MentionDropdownProps
->(({ query, clientRect, command, onClose, owner, conversationId }, ref) => {
-  const [selectedIndex, setSelectedIndex] = useState(0);
-  const triggerRect = useMemo(
-    () => (clientRect ? clientRect() : null),
-    [clientRect]
-  );
-
-  // Fetch suggestions from server using the query.
-  const { suggestions, isLoading } = useMentionSuggestions({
-    workspaceId: owner.sId,
-    conversationId,
-    query,
-    select: { agents: true, users: true },
-  });
-
-  const triggerRef = useRef<HTMLDivElement>(null);
-  const [virtualTriggerStyle, setVirtualTriggerStyle] =
-    useState<React.CSSProperties>({});
-
-  const selectItem = (index: number) => {
-    const item = suggestions[index];
-
-    if (item) {
-      command(item);
-    }
-  };
-
-  const updateTriggerPosition = useCallback(() => {
-    if (triggerRect && triggerRef.current) {
-      setVirtualTriggerStyle({
-        position: "fixed",
-        left: triggerRect.left,
-        // On iOS based browsers, the position is not correct without adding the offsetTop.
-        // Something related to the position calculation when there is a scrollable area.
-        top: triggerRect.top + (window.visualViewport?.offsetTop ?? 0),
-        width: 1,
-        height: triggerRect.height || 1,
-        pointerEvents: "none",
-        zIndex: -1,
-      });
-    }
-  }, [triggerRect]);
-
-  useImperativeHandle(ref, () => ({
-    onKeyDown: ({ event }) => {
-      if (event.key === "ArrowUp") {
-        setSelectedIndex(
-          (selectedIndex + suggestions.length - 1) % suggestions.length
-        );
-        return true;
+>(
+  (
+    {
+      query,
+      clientRect,
+      command,
+      onClose,
+      owner,
+      conversationId,
+      preferredAgentId,
+      userMentionsEnabled,
+    },
+    ref
+  ) => {
+    const [selectedIndex, setSelectedIndex] = useState(0);
+    const triggerRect = useMemo(
+      () => (clientRect ? clientRect() : null),
+      [clientRect]
+    );
+
+    // Fetch suggestions from server using the query.
+    const { suggestions, isLoading } = useMentionSuggestions({
+      workspaceId: owner.sId,
+      conversationId,
+      query,
+      select: { agents: true, users: userMentionsEnabled ?? false },
```

Comment:
you can keep true, there is a check backend side in `front/pages/api/w/[wId]/assistant/conversations/[cId]/mentions/suggestions.ts`